### PR TITLE
fix(text-field): make sure validation is not eager

### PR DESCRIPTION
**Pull request by YonatanKra**

### CODE REVIEW

File: libs/components/src/lib/calendar-event/ui.test.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {
 
 const components = ['calendar', 'calendar-event'];
 
-test.only('should show the component', async ({ page }: { page: Page }) => {
+test('should show the component', async ({ page }: { page: Page }) => {
```
Comment (yinonov) [inline libs/components/src/lib/calendar-event/ui.test.ts:10]: sorry 😬

File: libs/components/src/lib/text-field/text-field.ts [hunk 2]
```diff
@@ -24,18 +24,31 @@ export class TextField extends FoundationTextfield {
...
+			this.#blurred = false;
+		});
+	}
+
+	override validate = () => {
 		super.validate();
+		this.userValid = !this.userValid;
 		if (this.proxy instanceof HTMLElement) {
-			this.userValid = this.dirtyValue ? !this.validationMessage : true;
+			this.userValid = (this.#blurred && this.dirtyValue) ? !this.validationMessage : true;
```
Comment (yinonov) [inline libs/components/src/lib/text-field/text-field.ts:49]: this assigns it wrong according to https://developer.mozilla.org/en-US/docs/web/css/:user-valid.
I think the composite condition should be assigned to a `reflectInvalid` flag instead

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/text-field/text-field.ts (hunk 2 above)
Comment (YonatanKra) [reply libs/components/src/lib/text-field/text-field.ts:49]: @yinonov - this actually follows the `user-valid` spec as we discussed.

Comment (yinonov) [reply libs/components/src/lib/text-field/text-field.ts:49]: yes we discussed it after this comment. resolving

File: libs/components/src/lib/text-field/text-field.spec.ts [hunk 3]
```diff
@@ -355,22 +365,27 @@ describe('vwc-text-field', () => {
 			element.validate();
 		}
 
-		it('should add class error to root', async function () {
+		it('should add class error to base if invalid', async function () {
```
Comment (yinonov) [inline libs/components/src/lib/text-field/text-field.spec.ts:368]: maybe the intent is wrong. maybe it should be adding `user-not-valid` class to be completely correct

Comment (yinonov) [reply libs/components/src/lib/text-field/text-field.spec.ts:368]: notice it is `not-valid` rather than `invalid`

Comment (YonatanKra) [reply libs/components/src/lib/text-field/text-field.spec.ts]: @yinonov 
Looked a bit deeper.
Do we really need the `user-not-valid` class?
We already have the semantic `error` error class that speaks our css design system language.

**Pull request merged**